ros2topic/test/test_echo_pub.py

```diff
@@ -494,9 +519,9 @@ def publish_message():
                 )
                 assert command.wait_for_output(functools.partial(
                     launch_testing.tools.expect_output, expected_lines=[
-                        "b'\\x00\\x01\\x00\\x00\\x06\\x00\\x00\\x00hello\\x00\\x00\\x00'",
-                        '---',
-                    ], strict=True
+                        re.compile(r"^b'\\x00\\x01\\x00\\x00\\x06\\x00\\x00\\x00hello(\\x00)*'$"),
+                        re.compile(r'^---$'),
```

Comment:
with a regular expression pattern that matches the same string but allows for any number of `\x00` characters at the end. This makes the matching less strict and more flexible. can you explain why this change is required to add here? rmw_zenoh prints the less `\x00` characters at the end related to the following fix below?

Reply:
yes, DDS is returning `'\\x00\\x01\\x00\\x00\\x06\\x00\\x00\\x00hello\\x00\\x00\\x00` and zenoh `\\x00\\x01\\x00\\x00\\x06\\x00\\x00\\x00hello\\x00`, 2 less `\\x00` at the end.

The number of bytes changed from 16 to 14 , as you can see in the other tests